packages/approval-controller/src/ApprovalController.ts

```diff
@@ -768,7 +768,9 @@ export class ApprovalController extends BaseControllerV2<
       draftState.approvalFlows.push({ id, loadingText });
     });
 
-    this.#showApprovalRequest();
+    if (!opts.showOnFirstRequest) {
```

Comment:
Apologies as I suggested this name, but on reflection it's not guaranteed to show on the first request either as they may choose to set `shouldShowRequest` to `false` or use the `add` method directly rather than `addAndShowApprovalRequest`.

To be consistent with `shouldShowRequest` in the `addRequest` message listener, would simply `show` be the best name for this option and we default to `true` so the behaviour doesn't change.

As a result, we'd have to change this condition to `opts.show !== false` so if the option isn't specified we still get our default behaviour.

Reply:
Can we add this option to the JSDoc for this method?

Reply:
Done!